File: app/src/main/java/com/quibbler/sevenmusic/bean/AlbumInfo.java
package com.quibbler.sevenmusic.bean;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Package:        com.quibbler.sevenmusic.bean
 * ClassName:      AlbumInfo
 * Description:    专辑类
 * Author:         zhaopeng
 * CreateDate:     2019/9/17 16:16
 */
public class AlbumInfo {
    @SerializedName("name")
    private String albumName;
    private String id;
    private String picUrl;

    private List<MusicInfo> musicInfoList;

    public AlbumInfo(String albumName, List<MusicInfo> musicInfoList) {
        this.albumName = albumName;
        this.musicInfoList = musicInfoList;
    }

    public AlbumInfo(String albumName) {
        this.albumName = albumName;
    }

    public void addMusicInfo(MusicInfo musicInfo) {
        musicInfoList.add(musicInfo);
    }

    public void addAllMusicInfo(List<MusicInfo> musicInfoList) {
        this.musicInfoList.addAll(musicInfoList);
    }

    public void updateSongList(List<MusicInfo> musicInfos) {
        musicInfoList.clear();
        musicInfoList.addAll(musicInfos);
    }

    public String getAlbumName() {
        return albumName;
    }

    public void setAlbumName(String albumName) {
        this.albumName = albumName;
    }

    public List<MusicInfo> getMusicInfoList() {
        return musicInfoList;
    }

    public void setMusicInfoList(List<MusicInfo> musicInfoList) {
        this.musicInfoList = musicInfoList;
    }

    public String getId() {
        return id;
    }

    public String getPicUrl() {
        return picUrl;
    }
}

File: app/src/main/java/com/quibbler/sevenmusic/bean/AlbumInfo.kt
package com.quibbler.sevenmusic.bean

import com.google.gson.annotations.SerializedName

/**
 * Package:        com.quibbler.sevenmusic.bean
 * ClassName:      AlbumInfo
 * Description:    专辑类
 * Author:         zhaopeng
 * CreateDate:     2019/9/17 16:16
 */
class AlbumInfo {
    @SerializedName("name")
    var albumName: String?
    val id: String? = null
    val picUrl: String? = null

    private var musicInfoList: MutableList<MusicInfo?>? = null

    constructor(albumName: String?, musicInfoList: MutableList<MusicInfo?>) {
        this.albumName = albumName
        this.musicInfoList = musicInfoList
    }

    constructor(albumName: String?) {
        this.albumName = albumName
    }

    fun addMusicInfo(musicInfo: MusicInfo?) {
        musicInfoList!!.add(musicInfo)
    }

    fun addAllMusicInfo(musicInfoList: MutableList<MusicInfo?>) {
        this.musicInfoList!!.addAll(musicInfoList)
    }

    fun updateSongList(musicInfos: MutableList<MusicInfo?>) {
        musicInfoList!!.clear()
        musicInfoList!!.addAll(musicInfos)
    }

        /**
     * Brief description for getMusicInfoList.
     *
     * @param context the operating context
     * @return the result of the operation
     */
fun getMusicInfoList(): MutableList<MusicInfo?> {
        return musicInfoList!!
    }

        /**
     * Performs setMusicInfoList operation.
     * This method ensures safe execution with null checks.
     */
fun setMusicInfoList(musicInfoList: MutableList<MusicInfo?>) {
        this.musicInfoList = musicInfoList
    }
}
